components/event-sources/upgrade-job/process/save_current_state_test.go

```diff
@@ -147,4 +148,63 @@ func TestSaveCurrentState(t *testing.T) {
 		g.Expect(err).Should(gomega.BeNil())
 		g.Expect(namespaces.Items).To(gomega.Equal(e2eSetup.namespaces.Items))
 	})
+
+	t.Run("Save the current state without a config map when application list is empty", func(t *testing.T) {
+		e2eSetup.applications = &applicationconnectorv1alpha1.ApplicationList{}
+		p := &Process{
+			ReleaseName:     "release",
+			BEBNamespace:    "ns",
+			EventTypePrefix: "prefix",
+			Logger:          logrus.New(),
+		}
+		p.Clients = getProcessClients(e2eSetup, g)
+		saveCurrentStateStep := NewSaveCurrentState(p)
+		expectedLabelsInCM := map[string]string{
+			"release": p.ReleaseName,
+		}
+		p.Steps = []Step{
+			saveCurrentStateStep,
+		}
+		err := p.Execute()
+		g.Expect(err).Should(gomega.BeNil())
+
```

Comment:
Does it make sense to check `g.Expect(len(p.State.Applications.Items)).To(gomega.Equal(0))`?

Reply:
Actually, we are testing the contents of the `configmap`(which stores the state and does not contain applications) and not the state of `p`. Hence the state of `p` is not valuable in this context, IMO.